GoogleDataTransport/GDTCORTests/Unit/GDTCORStorageTest.m

```diff
@@ -442,4 +442,27 @@ - (void)testQoSTierFast {
   });
 }
 
+/** Tests storing events at the same time an appWillTerminate notif is being sent. */
+- (void)testStoringEventsDuringTerminate {
+  int numberOfIterations = 1000;
+  for (int i = 0; i < numberOfIterations; i++) {
+    NSString *testString = [NSString stringWithFormat:@"testString %d", i];
+    GDTCOREvent *event = [[GDTCOREvent alloc] initWithMappingID:@"404" target:target];
+    event.dataObjectTransportBytes = [testString dataUsingEncoding:NSUTF8StringEncoding];
+    event.clockSnapshot = [GDTCORClock snapshot];
+    XCTestExpectation *writtenExpectation = [self expectationWithDescription:@"event written"];
+    XCTAssertNoThrow([[GDTCORStorage sharedInstance] storeEvent:event
+                                                     onComplete:^(BOOL wasWritten, NSError *error) {
+                                                       XCTAssertTrue(wasWritten);
+                                                       [writtenExpectation fulfill];
+                                                     }]);
+    [self waitForExpectationsWithTimeout:10 handler:nil];
```

Comment:
I wonder if this test actually testing the fix? Looks like `UIApplicationWillTerminateNotification` won't be posted until `writtenExpectation` is fulfilled. Could you please add clarifying comments if the test is actually correct.

Reply:
The test fails without the change. Added a comment.

Reply:
Thank you for adding the comment. 
nit: It was still a bit unclear to me how the test creates concurrency until I deep dived into the tested class implementation  details. I think it would be nice to add comments explaining that the notification triggers  storing events by GDTCORStorage which my concurrently access ivars on the next `[GDTCORStorage storeEvent:...]` invocation.